### DISCUSSION TITLE
Seeking Advice: Optimal Integration of Firebase with Nylo

### DISCUSSION THREAD

**Discussion by fredrickreuben** [nylo-core/nylo#61] [General] [answered]
"Hey everyone, I'm currently working on a project where I need to integrate Firebase with Nylo. I've been researching different approaches, but I'm wondering what the ideal way is to achieve this integration. If anyone has experience or insights into best practices for connecting Firebase with Nylo, I'd greatly appreciate your guidance! Thanks in advance!"

**Comment by agordn52** [comment, marked as answer]
Hi @fredrickreuben,

You could integrate Firebase as per the docs on their site https://firebase.flutter.dev/docs/overview/

I'd recommend calling the `initializeApp` method inside a provider.
You can create a provider by running this in the terminal.

``` bash
dart run nylo_framework:main make:provider firebase
```

Then, open the file **lib/app/providers/firebase_provider.dart**

Add the `initializeApp` method and any other Firebase-related code in the provider. Providers are booted at the start of your application.

``` dart
import 'package:nylo_framework/nylo_framework.dart';

class FirebaseProvider implements NyProvider {
  @override
  boot(Nylo nylo) async {
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );
...
```

I hope that makes sense, I think I'll plan a video around this for the YouTube [channel](https://www.youtube.com/@nylo_dev), it may help others too.
